cognite/client/utils/_time.py

```diff
@@ -9,59 +9,111 @@
-
-
+UTC = ZoneInfo("UTC")  # type: ignore [abstract]
 UNIT_IN_MS_WITHOUT_WEEK = {"s": 1000, "m": 60000, "h": 3600000, "d": 86400000}
 UNIT_IN_MS = {**UNIT_IN_MS_WITHOUT_WEEK, "w": 604800000}
 VARIABLE_LENGTH_UNITS = {"month", "quarter", "year"}
 GRANULARITY_IN_HOURS = {"w": 168, "d": 24, "h": 1}
 GRANULARITY_IN_TIMEDELTA_UNIT = {"w": "weeks", "d": "days", "h": "hours", "m": "minutes", "s": "seconds"}
 MIN_TIMESTAMP_MS = -2208988800000  # 1900-01-01 00:00:00.000
 MAX_TIMESTAMP_MS = 4102444799999  # 2099-12-31 23:59:59.999
+_GRANULARITY_UNIT_LOOKUP: dict[str, str] = {
+    "s": "s",
+    "sec": "s",
+    "second": "s",
+    "seconds": "s",
+    "t": "m",
+    "m": "m",
+    "min": "m",
+    "minute": "m",
+    "minutes": "m",
+    "h": "h",
+    "hour": "h",
+    "hours": "h",
+    "d": "d",
+    "day": "d",
+    "days": "d",
+    "w": "w",
+    "week": "w",
+    "weeks": "w",
+    "mo": "month",
+    "month": "month",
+    "months": "month",
+    "q": "quarter",
+    "quarter": "quarter",
+    "quarters": "quarter",
+    "y": "year",
+    "year": "year",
+    "years": "year",
+}
+_GRANULARITY_CONVERSION = {
+    "s": (1, "s"),
+    "m": (1, "m"),
+    "h": (1, "h"),
+    "d": (1, "d"),
+    "w": (7, "d"),
+    "month": (1, "mo"),
+    "quarter": (3, "mo"),
+    "year": (12, "mo"),
+}
 
 
-def import_zoneinfo() -> type[ZoneInfo]:
-    try:
-        if sys.version_info >= (3, 9):
-            from zoneinfo import ZoneInfo
-        else:
-            from backports.zoneinfo import ZoneInfo
-        return ZoneInfo
-
-    except ImportError as e:
-        from cognite.client.exceptions import CogniteImportError
-
-        raise CogniteImportError(
-            "ZoneInfo is part of the standard library starting with Python >=3.9. In earlier versions "
-            "you need to install a backport. This is done automatically for you when installing with the pandas "
-            "group: 'cognite-sdk[pandas]', or with poetry: 'poetry install -E pandas'"
-        ) from e
-
-
-def _import_zoneinfo_not_found_error() -> type[ZoneInfoNotFoundError]:
-    if sys.version_info >= (3, 9):
-        from zoneinfo import ZoneInfoNotFoundError
-    else:
-        from backports.zoneinfo import ZoneInfoNotFoundError
-    return ZoneInfoNotFoundError
+def parse_str_timezone_offset(tz: str) -> timezone:
+    """
+    This function attempts to accept and convert all valid fixed-offset timezone input that the API
+    supports for datapoints endpoints. The backend is using native java class TimeZone with some
+    added restrictions on ambiguous names/ids.
+    """
+    prefix, tz = "", tz.replace(" ", "")
+    if match := re.match("^(UTC?|GMT)?", tz):
+        tz = tz.replace(prefix := match.group(), "")
+    if prefix and not tz:
+        return timezone.utc
+    elif re.match(r"^(-|\+)\d\d?$", tz) and abs(hours_offset := int(tz)) <= 18:
+        return timezone(timedelta(hours=hours_offset))
+    return cast(timezone, datetime.strptime(tz, "%z").tzinfo)
```

Comment:
Can't run this code in my head, so just want to check, we do support UTC+05:30 and similar? (Is that what datetime.strptime does?)

Reply:
I assume this is only called when we couldn't parse time zones like "Europe/Oslo", so that we don't have to handle those.

Reply:
Yup, `"%z"` parses any UTC offset, so I'm relying as much as possible on built-in datetime parsing